openapi/plugin_config_services_test.go

```diff
@@ -46,3 +64,67 @@ func TestPluginConfigSchemaV1IsSecureSkipVerifyEnabled(t *testing.T) {
 		})
 	})
 }
+
+func TestServiceConfigV1Validate(t *testing.T) {
+	Convey("Given a ServiceConfigV1 containing a valid swagger URL and a specific plugin version", t, func() {
+		var serviceConfiguration ServiceConfiguration
+		expectedPluginVersion := "0.14.0"
+		expectedSwaggerURL := "http://sevice-api.com/swagger.yaml"
+		serviceConfiguration = &ServiceConfigV1{
+			SwaggerURL:    expectedSwaggerURL,
+			PluginVersion: expectedPluginVersion,
+		}
+		Convey("When Validate method is called with a running version that matches the configured one", func() {
+			runningPluginVersion := expectedPluginVersion
+			err := serviceConfiguration.Validate(runningPluginVersion)
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+		})
+	})
+	Convey("Given a ServiceConfigV1 containing an invalid swagger URL pointing at a file store in the disk", t, func() {
+		var serviceConfiguration ServiceConfiguration
+		swaggerFile, _ := ioutil.TempFile("", "")
+		defer os.RemoveAll(swaggerFile.Name()) // clean up
```

Comment:
missing file handle closure - best to defer the call to remove and swaggerFile.Close() https://golang.org/pkg/io/ioutil/#example_TempFile